new-log-viewer/src/components/MenuBar.tsx

```diff
@@ -0,0 +1,210 @@
+            updateWindowUrlHashParams({logEventNum: numEvents});
+            break;
+        default:
+            break;
+    }
+};
+
+/**
+ *
+ */
+export const MenuBar = () => {
+    const {setMode, mode} = useColorScheme();
+    const {logEventNum} = useContext(UrlContext);
+    const {fileName, loadFile, numEvents} = useContext(StateContext);
+
+    const [settingsModelOpen, setSettingsModelOpen] = useState<boolean>(false);
+
+    const handleNavButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        if (null === logEventNum) {
+            return;
+        }
+        const {actionName} = event.currentTarget.dataset as { actionName: ACTION_NAME };
+        if (Object.values(ACTION_NAME).includes(actionName)) {
+            handleAction(actionName, logEventNum, numEvents);
+        }
+    };
+
+    const handleOpenFileButtonClick = () => {
+        openFile((file) => {
+            loadFile(file, {code: CURSOR_CODE.LAST_EVENT, args: null});
+        });
+    };
+
+    const SmallNavIconButton = ({actionName, Icon}: {
+        actionName: string,
+        Icon: SvgIconComponent,
+    }) => (
+        <IconButton
+            data-action-name={actionName}
+            size={"sm"}
+            onClick={handleNavButtonClick}
+        >
+            <Icon/>
+        </IconButton>
+    );
+    const SmallIconButton = ({onClick, Icon}: {
+        onClick: (event: React.MouseEvent<HTMLButtonElement>) => void,
+        Icon: SvgIconComponent,
+    }) => (
+        <IconButton
+            size={"sm"}
+            onClick={onClick}
+        >
+            <Icon/>
+        </IconButton>
+    );
+
+    return (
+        <Sheet className={"menu-bar"}>
+            <Typography className={"menu-bar-typography"}>
+                <Description/>
+                {fileName}
+            </Typography>
+
+            <SmallNavIconButton
+                actionName={ACTION_NAME.FIRST_PAGE}
+                Icon={SkipPrevious}/>
+            <SmallNavIconButton
+                actionName={ACTION_NAME.PREV_PAGE}
+                Icon={NavigateBefore}/>
+            <SmallNavIconButton
+                actionName={ACTION_NAME.NEXT_PAGE}
+                Icon={NavigateNext}/>
+            <SmallNavIconButton
+                actionName={ACTION_NAME.LAST_PAGE}
+                Icon={SkipNext}/>
+            <SmallIconButton
+                Icon={FileOpenIcon}
+                onClick={handleOpenFileButtonClick}/>
+            <SmallIconButton
+                Icon={Settings}
+                onClick={() => {
+                    setSettingsModelOpen(true);
+                }}/>
+            <IconButton
+                size={"sm"}
+            >
+                <TipsAndUpdates/>
+            </IconButton>
+            <Modal
+                open={settingsModelOpen}
+                onClose={() => {
+                    setSettingsModelOpen(false);
+                }}
+            >
+                <ModalDialog>
```

Comment:
Shall we make it fullscreen?
```suggestion
                <ModalDialog layout={"fullscreen"}>
```

Reply:
Well, it turns out fullscreen is overwhelming because it takes too much space for only a few info. I think reverting this back will be better.
<img width="1920" alt="image" src="https://github.com/user-attachments/assets/200e7e8e-f467-44d1-8e0d-8d9f0cd4f8ad">

Reply:
Sorry I might have missed this in the previous comments:
Why do we need `orientation={"horizontal"}` on the `FormControl` component?